samza-core/src/main/java/org/apache/samza/execution/LocalJobPlanner.java

```diff
@@ -107,10 +107,10 @@ private void createStreams(String planId, List<StreamSpec> intStreams, StreamMan
     LOG.info("A single processor must create the intermediate streams. Processor {} will attempt to acquire the lock.", uid);
     // Move the scope of coordination utils within stream creation to address long idle connection problem.
     // Refer SAMZA-1385 for more details
-    JobCoordinatorConfig jcConfig = new JobCoordinatorConfig(config);
-    String coordinationId = new ApplicationConfig(config).getGlobalAppId() + APPLICATION_RUNNER_PATH_SUFFIX;
+    JobCoordinatorConfig jcConfig = new JobCoordinatorConfig(userConfig);
```

Comment:
Orthogonal question: Does intermediate stream creation need anything from the generated configs?

Reply:
Yep, I just wanted to make sure that the usage of the `userConfig` here does not need the generated config. Thanks for clarifying that actual intermediate stream creation does use generated configs later on.